tests/unit_tests/seraphis_cache.cpp

```diff
@@ -0,0 +1,74 @@
+#include "seraphis_cache.h"
+
+#include <boost/range/adaptor/reversed.hpp>
+#include <gtest/gtest.h>
+
+bool HalfPruning::should_prune(const std::deque<uint64_t> &window, uint64_t chain_top)
+{
+    if (chain_top - *window.end() <= m_num_unprunable)
+        return false;
+
+    uint64_t width = *window.end() - *window.begin();
+    return window.size() * m_max_separation > width;
+}
+
+uint64_t Cache::get_nearest_block_height_clampdown(uint64_t test_height)
```

Comment:
```suggestion
uint64_t Cache::get_nearest_block_height_clampdown(const uint64_t test_height)
```

Mark parameters `const` if they are supposed to be immutable.